Octokit/Clients/TeamsClient.cs

```diff
@@ -37,6 +38,28 @@ public Task<Team> Get(int id)
             return ApiConnection.Get<Team>(endpoint, null, AcceptHeaders.NestedTeamsPreview);
         }
 
+        /// <summary>
+        /// Gets a single <see cref="Team"/> by name.
+        /// Note: this is more expensive than using the
+        /// team ID, as first we perform a lookup to determine
+        /// matching teams.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/orgs/teams/#get-team
+        /// </remarks>
+        /// <param name="orgName">The organization name.</param>
+        /// <param name="teamName">The team name.</param>
+        /// <returns>The <see cref="Team"/> with the given identifier.</returns>
+        public async Task<Team> Get(string orgName, string teamName)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(orgName, nameof(orgName));
+            Ensure.ArgumentNotNullOrEmptyString(teamName, nameof(teamName));
+
+            var allTeams = await this.GetAll(orgName);
+
+            return allTeams.First(x => string.Equals(x.Name, teamName, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
ℹ️ I think `CurrentCultureIgnoreCase` is acceptable here but want to call attention to it just in case.